### PR TITLE
Homework2: Alex Levdansky

### PR DESCRIPTION
**Pull request by Coromilly**

### CODE REVIEW

File: Homework1.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+import random
+import re
+import sys
+from time import localtime
+import unittest
+import xml.etree.ElementTree as Etree
+
+# 1
+print('Hello, world!')
+# 2
```
Comment (belovmd) [inline Homework1.py:16]: It would be much more readable if we separate tasks with two empty lines

File: Homework1.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+parents, babies = (1, 1)
+while babies < 100:
+    print('This generation has {0} babies'.format(babies))
+    parents, babies = (babies, parents + babies)
+
+
+# 5
+def greet(nane):
+    print('Hello', nane)
+
```
Comment (belovmd) [inline Homework1.py:37]: no sense to separate with two lines function and it's calls in such a big file

File: Homework1.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+my_purchase = {
+    'apple': 1,
+    'banana': 6}
+grocery_bill = sum(prices[fruit] * my_purchase[fruit]
+                   for fruit in my_purchase)
+print('I owe the grocer $%.2f' % grocery_bill)
+# 8
+# This program adds up integers that have been passed
+# as arguments in the command line
+
```
Comment (belovmd) [inline Homework1.py:65]: it looks strange our docstring goes after previous without empty line and separated from its solution.

**Review by belovmd — CHANGES_REQUESTED** [review]
Please reformat this code a bit to make it a bit more readable

File: fizz_buzz.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+for number in range(101):
+    if number % 3 == 0 and number % 5 == 0:
```
Comment (belovmd) [inline fizz_buzz.py:2]: you can use 15 here instead

File: money_calculation.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+price = str(input('Input price, separate dollars from cents with "." : '))
+num_of_items = int(input('Input number of items you want to buy: '))
+dollars = int(price.split('.')[0])
+cents = int(price.split('.')[1])
+print('One item price is {0} dollars {1} cents'.format(dollars, cents))
```
Comment (belovmd) [inline money_calculation.py:5]: In description wasn't mentioned this print

File: money_calculation.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+if cents * num_of_items >= 100:
```
Comment (belovmd) [inline money_calculation.py:6]: we can avoid this if condition here

File: money_calculation.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+price = str(input('Input price, separate dollars from cents with "." : '))
```
Comment (belovmd) [inline money_calculation.py:1]: I will enter 12.23423141234

File: task3.py [hunk 8]
```diff
@@ -0,0 +1,10 @@
+def checkio(number: int) -> int:
+    """Calculate the product of the digits excluding any zeroes."""
```
Comment (belovmd) [inline task3.py:2]: Please put more clear description here

File: task5.py [hunk 9]
```diff
@@ -0,0 +1,10 @@
+def checkio(array):
+    """sums even-indexes elements and multiply at the last."""
+    sum = 0
+    if array == []:
```
Comment (belovmd) [inline task5.py:4]: `if not array` would be enough?

**Review by belovmd — CHANGES_REQUESTED** [review]
1. please remove files from HW1
2. Please solve Palindrome task without strings methods - only using arithmetical operations
3. Please address these issues:

File: task5.py (hunk 9 above)
Comment (Coromilly) [reply task5.py:4]: Solved.

File: task3.py (hunk 8 above)
Comment (Coromilly) [reply task3.py:2]: fixed.

File: fizz_buzz.py (hunk 4 above)
Comment (Coromilly) [reply fizz_buzz.py:2]: fixed.

File: money_calculation.py (hunk 5 above)
Comment (Coromilly) [reply money_calculation.py:5]: Fixed

File: money_calculation.py (hunk 7 above)
Comment (Coromilly) [reply money_calculation.py:1]: solved.

File: money_calculation.py (hunk 6 above)
Comment (Coromilly) [reply money_calculation.py:6]: solved.

**Pull request closed without merging**